reframe/core/schedulers/slurm.py

```diff
@@ -345,11 +360,13 @@ def filternodes(self, job, nodes):
             nodes &= self._get_reservation_nodes(reservation)
             self.log(f'[F] Filtering nodes by reservation {reservation}: '
                      f'available nodes now: {len(nodes)}')
-
         if partitions:
             partitions = set(partitions.strip().split(','))
         else:
-            default_partition = self._get_default_partition()
+            default_partition = (
```

Comment:
Maybe add a short comment inside the `else` branch explaining this concept.

Reply:
Thanks, I've added a comment.